packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx

```diff
@@ -878,11 +914,28 @@ function MapAnalysisImpl(props: Props & CompleteAppState) {
   return (
     <PromiseResult state={appsPromiseState}>
       {(apps: ComputationAppOverview[]) => {
-        const activeSideNavigationItemMenu =
-          activeSideMenuIndex != null &&
-          sideNavigationButtonConfigurationObjects[
-            activeSideMenuIndex
-          ].renderSideNavigationPanel(apps);
+        const activeSideNavigationItemMenu = getSideNavigationItemMenu();
+
+        function getSideNavigationItemMenu() {
+          if (activeSideMenuIndex == null) return <></>;
+          // if activeSideMenuIndex does not correspond to an object, then the activeSideMenuIndex must
+          // correspond to a subMenuItem's index
+          else if (
+            sideNavigationButtonConfigurationObjects[activeSideMenuIndex]
+          ) {
+            return sideNavigationButtonConfigurationObjects[
+              activeSideMenuIndex
+            ].renderSideNavigationPanel(apps);
+          } else {
+            // "normalize" the activeSideMenuIndex to resolve the parent's index
+            const indexOfParentContainer = Math.floor(
+              activeSideMenuIndex / 10 - 1
+            );
+            return sideNavigationButtonConfigurationObjects[
+              indexOfParentContainer
+            ].renderSideNavigationPanel(apps);
+          }
+        }
```

Comment:
I'm not crazy about how this is being done. Instead of using indexes, I think we should give each menu and submenu item a unique id, and use that to keep track of which menu item is selected. If a top-level menu item has a submenu, then we should never set it as active. How does this sound?